### PR TITLE
Add a default workflow for updating the imported version

### PR DESCRIPTION
**Pull request by ahockersten**
This updates from upstream once per hour.

### CODE REVIEW

**Review by bittrance — APPROVED** [review]

File: .github/workflows/update-azure-devops-templates-from-upstream.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Update the main branch to what upstream has on an hourly basis (and when triggered manually).
+# This is to ensure local copies always match upstream as close as possible.
+name: update-from-upstream
+
+on:
+  schedule:
+    - cron: "0 * * * *" # every hour
+  workflow_dispatch:
```
Comment (bittrance) [inline .github/workflows/update-azure-devops-templates-from-upstream.yaml:8]: I think it is good practice to add an explicit `nil`.

File: .github/workflows/update-azure-devops-templates-from-upstream.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        # this makes no sense to run on the main repository, and it might not even work there
+        if: ${{ env.GITHUB_REPOSITORY != 'XenitAB/azure-devops-templates' }}
+        with:
+          ref: main
+          fetch-depth: 0
+      - run: |
+          set -e
+          git checkout main
+          git config --global user.email "github-workflows-@$GITHUB_WORKFLOW-$GITHUB_REPOSITORY"
+          git config --global user.name "Github Workflows ($GITHUB_WORKFLOW $GITHUB_REPOSITORY)"
```
Comment (bittrance) [inline .github/workflows/update-azure-devops-templates-from-upstream.yaml:25]: Do you need these for fetch/push?

**Review by bittrance — APPROVED** [review]
There is a section in the top-level readme about Azure DevOps. GitHub should prolly have a section too.

File: .github/workflows/update-azure-devops-templates-from-upstream.yaml (hunk 2 above)
Comment (ahockersten) [reply .github/workflows/update-azure-devops-templates-from-upstream.yaml:25]: No. Good point. I've removed this

File: .github/workflows/update-azure-devops-templates-from-upstream.yaml (hunk 1 above)
Comment (ahockersten) [reply .github/workflows/update-azure-devops-templates-from-upstream.yaml]: Done

**Pull request merged**